src/components/Publish/Pricing/Dynamic.tsx

```diff
@@ -50,10 +50,13 @@ export default function Dynamic({ content }: { content: any }): ReactElement {
   useEffect(() => {
     if (!accountId) {
       setError(`No account connected. Please connect your Web3 wallet.`)
+      values.user.hasBalance = false
```

Comment:
and why overwrite this in every check over and over again? There could be just one effect dedicated for checking the balance, and update `user.hasBalance` as needed instead of mixing this up with all the other checks

Reply:
setting the value field with `setFieldValue` introduces a side effect in the rendering. It kind of updates the field value after a couple of seconds, it's not really reactive. But probably based on what you said, it might not be the correct way to handle this. I added this new `hasBalance` parameter to the `user` object as I though It'd a the best place to track this, as the `pricing` object is mode of [`PriceOptions`](https://github.com/oceanprotocol/market/blob/fix/issue-1434-validate-balance-pricing-publish/src/@types/Price.d.ts#L54:L54). Also, logically I think this should be under the user's information, but I might be wrong.

Reply:
I end up removing `hasBalance` and handle this using effects instead.